from version 1.11.0 Vault MicroProfile Config Source stopped working

**Describe the bug**
from version 1.11.0 Vault MicroProfile Config Source stopped working to use vault with Databases (Vault MicroProfile Config Source) and OIDC

**Expected behavior**
quarkus fetched vault kv values first and then try verificted dependent properties. In version 1.10.5 works fine, but in 1.11.0 (and 1.11.1) does not work

**Actual behavior**
on startup throw error:
```
ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor#config threw an exception: java.util.NoSuchElementException: SRCFG00011: Could not expand value password in property quarkus.datasource.password
	at io.smallrye.config.ExpressionConfigSourceInterceptor.lambda$getValue$0(ExpressionConfigSourceInterceptor.java:44)
	at io.smallrye.common.expression.ExpressionNode.emit(ExpressionNode.java:22)
	at io.smallrye.common.expression.Expression.evaluateException(Expression.java:56)
	at io.smallrye.common.expression.Expression.evaluate(Expression.java:70)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:37)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:18)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
	at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:192)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:149)
	at io.smallrye.config.SmallRyeConfig.getOptionalValue(SmallRyeConfig.java:209)
	at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.config(ConfigEditorProcessor.java:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```


**To Reproduce**
in a new application in version 1.11.0 or 1.11.1 add vault, quarkus-reactive-pg-client, quarkus-resteasy, quarkus-resteasy-mutiny, quarkus-oidc, quarkus-vertx dependencies in appplication.properties add the following properties:

**Configuration**
```properties
quarkus.vertx.prefer-native-transport=true
quarkus.http.so-reuse-port=true
quarkus.http.tcp-quick-ack=true
quarkus.http.tcp-cork=true
quarkus.http.tcp-fast-open=true
quarkus.thread-pool.core-threads=5

quarkus.vault.url=http://localhost:8201
quarkus.vault.authentication.client-token=x.xxxxxxxxxxxxxx
quarkus.vault.secret-config-kv-path.database=database-product/credentials
quarkus.vault.secret-config-kv-path.oidc=identity-server/oidc

quarkus.datasource.db-kind=postgresql
quarkus.datasource.username=${database.username}
quarkus.datasource.password=${database.password}
quarkus.datasource.reactive.url=${database.url}
quarkus.datasource.reactive.max-size=20
quarkus.hibernate-orm.database.generation=none

quarkus.oidc.enabled=true
quarkus.oidc.application-type=service
quarkus.oidc.discovery-enabled=true
quarkus.oidc.auth-server-url=${oidc.auth-server-url}
quarkus.oidc.client-id=${oidc.client-id}

```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux 5.3.18-lp152.60-default x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: OpenJDK 64-Bit Server VM Corretto-11.0.10.9.1 (build 11.0.10+9-LTS, mixed mode)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.11.0 and 1.11.1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 3.6.3

yes thanks! the full log is:

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------< com.example.template:template-core >-------------
[INFO] Building template-core 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus-maven-plugin:1.11.1.Final:dev (default-cli) @ template-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 68 source files to /home/hernael/template/template-core/target/classes
Listening for transport dt_socket at address: 5005
2021-01-29 17:18:48,310 INFO  [org.jbo.threads] (main) JBoss Threads version 3.2.0.Final
2021-01-29 17:18:48,416 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor#config threw an exception: java.util.NoSuchElementException: SRCFG00011: Could not expand value database.url in property quarkus.datasource.reactive.url
	at io.smallrye.config.ExpressionConfigSourceInterceptor.lambda$getValue$0(ExpressionConfigSourceInterceptor.java:44)
	at io.smallrye.common.expression.ExpressionNode.emit(ExpressionNode.java:22)
	at io.smallrye.common.expression.Expression.evaluateException(Expression.java:56)
	at io.smallrye.common.expression.Expression.evaluate(Expression.java:70)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:37)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:18)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
	at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:192)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:149)
	at io.smallrye.config.SmallRyeConfig.getOptionalValue(SmallRyeConfig.java:209)
	at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.config(ConfigEditorProcessor.java:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:308)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:182)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:59)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:78)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:372)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:128)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:81)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor#config threw an exception: java.util.NoSuchElementException: SRCFG00011: Could not expand value database.url in property quarkus.datasource.reactive.url
	at io.smallrye.config.ExpressionConfigSourceInterceptor.lambda$getValue$0(ExpressionConfigSourceInterceptor.java:44)
	at io.smallrye.common.expression.ExpressionNode.emit(ExpressionNode.java:22)
	at io.smallrye.common.expression.Expression.evaluateException(Expression.java:56)
	at io.smallrye.common.expression.Expression.evaluate(Expression.java:70)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:37)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:18)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
	at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:192)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:149)
	at io.smallrye.config.SmallRyeConfig.getOptionalValue(SmallRyeConfig.java:209)
	at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.config(ConfigEditorProcessor.java:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:153)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:306)
	... 9 more
Caused by: java.util.NoSuchElementException: SRCFG00011: Could not expand value database.url in property quarkus.datasource.reactive.url
	at io.smallrye.config.ExpressionConfigSourceInterceptor.lambda$getValue$0(ExpressionConfigSourceInterceptor.java:44)
	at io.smallrye.common.expression.ExpressionNode.emit(ExpressionNode.java:22)
	at io.smallrye.common.expression.Expression.evaluateException(Expression.java:56)
	at io.smallrye.common.expression.Expression.evaluate(Expression.java:70)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:37)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:18)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
	at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:192)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:149)
	at io.smallrye.config.SmallRyeConfig.getOptionalValue(SmallRyeConfig.java:209)
	at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.config(ConfigEditorProcessor.java:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2021-01-29 17:18:48,416 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure

```

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
@@ -1,63 +1,40 @@
 package io.quarkus.vertx.http.deployment.devmode.console;
 
+import static io.quarkus.runtime.LaunchMode.DEVELOPMENT;
+
 import java.io.BufferedWriter;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.StandardOpenOption;
 import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Properties;
 
-import org.eclipse.microprofile.config.Config;
-import org.eclipse.microprofile.config.ConfigProvider;
-
+import io.quarkus.arc.runtime.ConfigRecorder;
+import io.quarkus.deployment.IsDevelopment;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.ConfigDescriptionBuildItem;
 import io.quarkus.dev.console.DevConsoleManager;
 import io.quarkus.devconsole.runtime.spi.DevConsolePostHandler;
 import io.quarkus.devconsole.spi.DevConsoleRouteBuildItem;
-import io.quarkus.devconsole.spi.DevConsoleTemplateInfoBuildItem;
+import io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem;
+import io.quarkus.runtime.configuration.ProfileManager;
+import io.quarkus.vertx.http.runtime.devmode.ConfigDescription;
+import io.quarkus.vertx.http.runtime.devmode.ConfigDescriptionsSupplier;
 import io.vertx.core.MultiMap;
 import io.vertx.ext.web.RoutingContext;
 
 public class ConfigEditorProcessor {
-
-    @BuildStep
-    DevConsoleTemplateInfoBuildItem config(List<ConfigDescriptionBuildItem> config) throws Exception {
-        List<CurrentConfig> configs = new ArrayList<>();
-        Config current = ConfigProvider.getConfig();
-
-        Properties appProperties = new Properties();
-        Path appProps = null;
-        for (Path i : DevConsoleManager.getHotReplacementContext().getResourcesDir()) {
-            Path app = i.resolve("application.properties");
-            if (Files.exists(app)) {
-                appProps = app;
-                break;
-            }
+    @BuildStep(onlyIf = IsDevelopment.class)
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public DevConsoleRuntimeTemplateInfoBuildItem config(ConfigRecorder recorder,
+            List<ConfigDescriptionBuildItem> configDescriptionBuildItems) {
+        List<ConfigDescription> configDescriptions = new ArrayList<>();
+        for (ConfigDescriptionBuildItem item : configDescriptionBuildItems) {
+            configDescriptions.add(
+                    new ConfigDescription(item.getPropertyName(), item.getDocs(), item.getDefaultValue()));
         }
-        if (appProps != null) {
-            try (InputStream in = Files.newInputStream(appProps)) {
-                appProperties.load(in);
-            }
-        }
-
-        for (ConfigDescriptionBuildItem i : config) {
-            if (i.getPropertyName().contains("*")) {
-                continue; //TODO: complex properties
-            }
-            configs.add(new CurrentConfig(i.getPropertyName(), i.getDocs(), i.getDefaultValue(),
-                    current.getOptionalValue(i.getPropertyName(), String.class).orElse(null),
-                    appProperties.getProperty(i.getPropertyName())));
-        }
-        Collections.sort(configs);
-
-        return new DevConsoleTemplateInfoBuildItem("config", configs);
+        return new DevConsoleRuntimeTemplateInfoBuildItem("config", new ConfigDescriptionsSupplier(configDescriptions));
     }
 
     @BuildStep
@@ -65,107 +42,49 @@ public class ConfigEditorProcessor {
         return new DevConsoleRouteBuildItem("config", "POST", new DevConsolePostHandler() {
             @Override
             protected void handlePost(RoutingContext event, MultiMap form) throws Exception {
-                String key = event.request().getFormAttribute("name");
+                String name = event.request().getFormAttribute("name");
                 String value = event.request().getFormAttribute("value");
 
-                Properties appProperties = new Properties();
-                Path appProps = null;
-                for (Path i : DevConsoleManager.getHotReplacementContext().getResourcesDir()) {
-                    Path app = i.resolve("application.properties");
-                    if (Files.exists(app)) {
-                        appProps = app;
+                List<Path> resourcesDir = DevConsoleManager.getHotReplacementContext().getResourcesDir();
+                if (resourcesDir.isEmpty()) {
+                    throw new IllegalStateException("Unable to manage configurations - no resource directory found");
+                }
+
+                // In the current project only
+                Path path = resourcesDir.get(0);
+                Path configPath = path.resolve("application.properties");
+                if (!Files.exists(configPath)) {
+                    configPath = Files.createFile(path.resolve("application.properties"));
+                }
+
+                String profile = ProfileManager.getActiveProfile();
+                name = !profile.equals(DEVELOPMENT.getDefaultProfile()) ? "%" + profile + "." + name : name;
+                List<String> lines = Files.readAllLines(configPath);
+                int nameLine = -1;
+                for (int i = 0, linesSize = lines.size(); i < linesSize; i++) {
+                    final String line = lines.get(i);
+                    if (line.startsWith(name)) {
+                        nameLine = i;
                         break;
                     }
                 }
-                boolean present = false;
-                if (appProps != null) {
-                    try (InputStream in = Files.newInputStream(appProps)) {
-                        appProperties.load(in);
-                        present = appProperties.containsKey(key);
-                    }
+
+                if (nameLine != -1) {
+                    lines.set(nameLine, name + "=" + value);
                 } else {
-                    // If there is no application.properties file then create a new one in the first module
-                    List<Path> resourcesDir = DevConsoleManager.getHotReplacementContext().getResourcesDir();
-                    if (resourcesDir.isEmpty()) {
-                        throw new IllegalStateException(
-                                "Unable to create application.properties - no resource directory found");
-                    }
-                    appProps = Files.createFile(resourcesDir.get(0).resolve("application.properties"));
+                    lines.add(name + "=" + value);
                 }
-                if (!present) {
-                    try (OutputStream out = Files.newOutputStream(appProps, StandardOpenOption.APPEND)) {
-                        out.write(("\n" + key + "=" + value).getBytes(StandardCharsets.UTF_8)); //TODO: escpaing
-                    }
-                } else {
-                    List<String> lines = Files.readAllLines(appProps);
-                    Iterator<String> it = lines.iterator();
-                    while (it.hasNext()) {
-                        String val = it.next();
-                        if (val.startsWith(key + "=")) {
-                            it.remove();
-                        }
-                    }
-                    lines.add(key + "=" + value);
-                    try (BufferedWriter writer = Files.newBufferedWriter(appProps)) {
-                        for (String i : lines) {
-                            writer.write(i);
-                            writer.newLine();
-                        }
+
+                try (BufferedWriter writer = Files.newBufferedWriter(configPath)) {
+                    for (String i : lines) {
+                        writer.write(i);
+                        writer.newLine();
                     }
                 }
+
                 DevConsoleManager.getHotReplacementContext().doScan(true);
                 flashMessage(event, "Configuration updated");
             }
         });
     }
-
-    public static class CurrentConfig implements Comparable<CurrentConfig> {
-        private final String propertyName;
-        private final String description;
-        private final String defaultValue;
-        private final String currentValue;
-        private final String appPropertiesValue;
-
-        public CurrentConfig(String propertyName, String description, String defaultValue, String currentValue,
-                String appPropertiesValue) {
-            this.propertyName = propertyName;
-            this.description = description;
-            this.defaultValue = defaultValue;
-            this.currentValue = currentValue;
-            this.appPropertiesValue = appPropertiesValue;
-        }
-
-        public String getPropertyName() {
-            return propertyName;
-        }
-
-        public String getDescription() {
-            return description;
-        }
-
-        public String getDefaultValue() {
-            return defaultValue;
-        }
-
-        public String getCurrentValue() {
-            return currentValue;
-        }
-
-        public String getAppPropertiesValue() {
-            return appPropertiesValue;
-        }
-
-        @Override
-        public int compareTo(CurrentConfig o) {
-            if (appPropertiesValue == null && o.appPropertiesValue != null) {
-                return 1;
-            }
-            if (appPropertiesValue != null && o.appPropertiesValue == null) {
-                return -1;
-            }
-
-            return propertyName.compareTo(o.propertyName);
-        }
-    }
-
 }

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
@@ -63,16 +63,22 @@ public class ConfigEditorProcessor {
                 int nameLine = -1;
                 for (int i = 0, linesSize = lines.size(); i < linesSize; i++) {
                     final String line = lines.get(i);
-                    if (line.startsWith(name)) {
+                    if (line.startsWith(name + "=")) {
                         nameLine = i;
                         break;
                     }
                 }
 
                 if (nameLine != -1) {
-                    lines.set(nameLine, name + "=" + value);
+                    if (value.isEmpty()) {
+                        lines.remove(nameLine);
+                    } else {
+                        lines.set(nameLine, name + "=" + value);
+                    }
                 } else {
-                    lines.add(name + "=" + value);
+                    if (!value.isEmpty()) {
+                        lines.add(name + "=" + value);
+                    }
                 }
 
                 try (BufferedWriter writer = Files.newBufferedWriter(configPath)) {

--- a/extensions/vertx-http/deployment/src/main/resources/dev-templates/io.quarkus.quarkus-vertx-http/config.html
+++ b/extensions/vertx-http/deployment/src/main/resources/dev-templates/io.quarkus.quarkus-vertx-http/config.html
@@ -13,25 +13,29 @@ color: gray;
         <th scope="col">Value</th>
         <th scope="col">Update</th>
         <th scope="col">Default Value</th>
+        <th scope="col">Source</th>
         <th scope="col">Description</th>
     </tr>
     </thead>
     <tbody>
     {#for item in info:config}
-    <tr class="{item.appPropertiesValue ? 'table-info' : ''}">
+    <tr>
         <form method="post" enctype="application/x-www-form-urlencoded">
-            <input type="hidden" name="name" value="{item.propertyName}"/>
+            <input type="hidden" name="name" value="{item.configValue.name}"/>
             <td>
-                {item.propertyName}
+                {item.configValue.name}
             </td>
             <td>
-                <input type="text" name="value" value="{item.currentValue}"/>
+                <input type="text" name="value" value="{item.configValue.value}"/>
             </td>
             <td>
                 <input type="submit" value="Update" >
             </td>
             <td>
                 {item.defaultValue}
+            </td>
+            <td>
+                {item.configValue.configSourceName}
             </td>
             <td>
                 {item.description.fmtJavadoc}

--- a/extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/devconsole/DevConsoleConfigEditorTest.java
+++ b/extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/devconsole/DevConsoleConfigEditorTest.java
@@ -36,4 +36,20 @@ public class DevConsoleConfigEditorTest {
 
     }
 
+    @Test
+    public void testSetEmptyValue() {
+        RestAssured.with()
+                .get("q/arc/beans")
+                .then()
+                .statusCode(200);
+        RestAssured.with().formParam("name", "quarkus.http.root-path").formParam("value", "")
+                .redirects().follow(false)
+                .post("q/dev/io.quarkus.quarkus-vertx-http/config")
+                .then()
+                .statusCode(303);
+        RestAssured.with()
+                .get("q/arc/beans")
+                .then()
+                .statusCode(200);
+    }
 }

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigDescription.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigDescription.java
@@ -1,0 +1,71 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import io.smallrye.config.ConfigValue;
+
+public class ConfigDescription implements Comparable<ConfigDescription> {
+    private String name;
+    private String description;
+    private String defaultValue;
+    private ConfigValue configValue;
+
+    public ConfigDescription() {
+    }
+
+    public ConfigDescription(final String name, final String description, final String defaultValue) {
+        this.name = name;
+        this.description = description;
+        this.defaultValue = defaultValue;
+    }
+
+    public ConfigDescription(
+            final String name,
+            final String description,
+            final String defaultValue,
+            final ConfigValue configValue) {
+        this.name = name;
+        this.description = description;
+        this.defaultValue = defaultValue;
+        this.configValue = configValue;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(final String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(final String description) {
+        this.description = description;
+    }
+
+    public String getDefaultValue() {
+        return defaultValue;
+    }
+
+    public void setDefaultValue(final String defaultValue) {
+        this.defaultValue = defaultValue;
+    }
+
+    public ConfigValue getConfigValue() {
+        return configValue;
+    }
+
+    public void setConfigValue(final ConfigValue configValue) {
+        this.configValue = configValue;
+    }
+
+    @Override
+    public int compareTo(ConfigDescription o) {
+        int ordinal = Integer.compare(o.configValue.getConfigSourceOrdinal(), this.configValue.getConfigSourceOrdinal());
+        if (ordinal == 0) {
+            return this.configValue.getName().compareTo(o.configValue.getName());
+        }
+        return ordinal;
+    }
+}

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigDescriptionsSupplier.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigDescriptionsSupplier.java
@@ -1,0 +1,58 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import io.smallrye.config.SmallRyeConfig;
+
+public class ConfigDescriptionsSupplier implements Supplier<List<ConfigDescription>> {
+    private List<ConfigDescription> configDescriptions;
+
+    public ConfigDescriptionsSupplier() {
+    }
+
+    public ConfigDescriptionsSupplier(final List<ConfigDescription> configDescriptions) {
+        this.configDescriptions = configDescriptions;
+    }
+
+    @Override
+    public List<ConfigDescription> get() {
+        List<String> properties = new ArrayList<>();
+        SmallRyeConfig current = (SmallRyeConfig) ConfigProvider.getConfig();
+
+        for (ConfigDescription item : configDescriptions) {
+            properties.add(item.getName());
+            item.setConfigValue(current.getConfigValue(item.getName()));
+        }
+
+        for (ConfigSource configSource : current.getConfigSources()) {
+            if (configSource.getName().equals("PropertiesConfigSource[source=Build system]")) {
+                properties.addAll(configSource.getPropertyNames());
+            }
+        }
+
+        for (String propertyName : current.getPropertyNames()) {
+            if (properties.contains(propertyName)) {
+                continue;
+            }
+
+            configDescriptions.add(new ConfigDescription(propertyName, null, null, current.getConfigValue(propertyName)));
+        }
+
+        Collections.sort(configDescriptions);
+        return configDescriptions;
+    }
+
+    public List<ConfigDescription> getConfigDescriptions() {
+        return configDescriptions;
+    }
+
+    public void setConfigDescriptions(final List<ConfigDescription> configDescriptions) {
+        this.configDescriptions = configDescriptions;
+    }
+}

